velox/functions/prestosql/tests/ArraysOverlapTest.cpp

```diff
@@ -0,0 +1,237 @@
+       {false, true, false},
+       {true, false, true},
+       {std::nullopt}});
+
+  auto expected = makeNullableFlatVector<bool>(
+      {true, true, true, false, true, true, true, true, std::nullopt});
+
+  testExpr(expected, "arrays_overlap(C0, C1)", {array1, array2});
+  testExpr(expected, "arrays_overlap(C1, C0)", {array1, array2});
+}
+
+/// Test inline strings.
+TEST_F(ArraysOverlapTest, shortStrings) {
+  using S = StringView;
+
+  auto array1 = makeNullableArrayVector<StringView>({
+      {"a"_sv, std::nullopt, S("b")},
+      {S("a"), S("b"), S("a"), S("a")},
+      {std::nullopt, S("b"), std::nullopt},
+      {"abcd"_sv},
+  });
+  auto array2 = makeNullableArrayVector<StringView>({
+      {S("a")},
+      {S("a"), S("b"), S("b")},
+      {std::nullopt, std::nullopt, std::nullopt},
+      {S("abc"), S("a"), S("b")},
+  });
+  auto expected =
+      makeNullableFlatVector<bool>({true, true, std::nullopt, false});
+  testExpr(expected, "arrays_overlap(C0, C1)", {array1, array2});
+  testExpr(expected, "arrays_overlap(C1, C0)", {array1, array2});
+}
+
+/// Test non-inline (> 12 length) strings.
+TEST_F(ArraysOverlapTest, longStrings) {
+  using S = StringView;
+
+  auto array1 = makeNullableArrayVector<StringView>({
+      {S("red shiny car ahead"), S("blue clear sky above")},
+      {std::nullopt,
+       S("blue clear sky above"),
+       S("yellow rose flowers"),
+       S("orange beautiful sunset")},
+      {},
+      {S("red shiny car ahead"),
+       S("purple is an elegant color"),
+       S("green plants make us happy")},
+  });
+  auto array2 = makeNullableArrayVector<StringView>({
+      {S("red shiny car ahead")},
+      {std::nullopt},
+      {},
+      {S("red shiny car ahead"), S("green plants make us happy")},
+  });
+  auto expected =
+      makeNullableFlatVector<bool>({true, std::nullopt, false, true});
+  testExpr(expected, "arrays_overlap(C0, C1)", {array1, array2});
+  testExpr(expected, "arrays_overlap(C1, C0)", {array1, array2});
+}
+
+//// When one of the arrays is constant.
+TEST_F(ArraysOverlapTest, constant) {
+  auto array1 = makeNullableArrayVector<int32_t>({
+      {1, -2, 3, std::nullopt, 4, 5, 6, std::nullopt},
+      {1, 2, -2, 1},
+      {3, 8, std::nullopt},
+      {3, 8, 5},
+      {1, 1, -2, -2, -2, 4, 8},
+  });
+  auto expected =
+      makeNullableFlatVector<bool>({true, true, std::nullopt, false, true});
+  testExpr(expected, "arrays_overlap(C0, ARRAY[1,-2,4])", {array1});
+  testExpr(expected, "arrays_overlap(ARRAY[1,-2,4], C0)", {array1});
+  testExpr(
+      expected, "arrays_overlap(ARRAY[1,1,-2,1,-2,4,1,4,4], C0)", {array1});
+
+  // Array containing NULLs.
+  expected = makeNullableFlatVector<bool>({
+      true,
+      true,
+      std::nullopt,
+      std::nullopt,
+      true,
+  });
+  testExpr(expected, "arrays_overlap(C0, ARRAY[1,NULL,4])", {array1});
+  testExpr(expected, "arrays_overlap(ARRAY[1,NULL,4], C0)", {array1});
+}
+
+TEST_F(ArraysOverlapTest, wrongTypes) {
```

Comment:
should we add a test for array dimension mismatch between the input arrays?

Reply:
E20220304 10:59:56.535413 729935360 Exceptions.h:68] /Users/karteek/workspace/velox/./velox/vector/SelectivityVector.h87setValidRangeend <= bits_.size() * sizeof(bits_[0]) * 8(1128481602 vs. 64)RUNTIMEINVALID_STATE
I20220304 11:00:58.165135 729935360 Expr.cpp:263] Inside: 8 from 0 to 8 wrap FLAT expr arrays_overlap(c0, c1)
unknown file: Failure
C++ exception with description "Exception: VeloxRuntimeError
Error Source: RUNTIME

It doesn't allow us create arrays with different lengths. The call fails before reaching arrays_overlap.